### ISSUE TITLE
Right way to handle ApolloErrors

### ISSUE THREAD

**Issue by EugeneDraitsev** [apollographql/react-apollo#1318]
Hi, I have one pretty simple, but important question about error handling in react-apollo. 

My problem is: I execute my query in GraphiQL and get an error in response, but also I have some data, that can be queried without error.

![untitled](https://user-images.githubusercontent.com/16540432/32364689-4197a4d8-c087-11e7-95f2-a7128d87fcda.png)

So I want to show to user this successfully queried data and show an error in section with error data, but in Apollo-connected component I can get only `errors` block, without successfully queried `data`

You can reproduced it like this:
```
{
  empireHero: hero(episode: 'works fine with this argument') {
    name
  }
  jediHero: hero(episode: 'failed with this argument') {
    name
  }
}
```
All you query will be 'invalid', instead of getting error and data sections, you will get only error section.
Probably I trying to make something wrong? Can you try help me with this problem?

**Version**
- apollo-client@ 2.0.1
- react-apollo@ 2.0.0

**Comment by peggyrayzis** [comment]
Hi @EugeneDraitsev! `react-apollo`'s default policy is to treat any GraphQL errors as runtime errors and to discard the data. To customize this policy, learn more [here](https://www.apollographql.com/docs/react/basics/queries.html#graphql-config-options-errorPolicy).

**Comment by edraitsev** [comment]
Hi @peggyrayzis, thanks for quick response. I tried to set my options like 
```
{
    variables: { appLang, tenant, manufacturers },
    errorPolicy: 'all', // Also tried all other values
}
```

But I still can't see any data, when query has errors (and doublechecked that I get data in GraphiQL)

When I get error in my `callQueue` and still can't see successfully fetched `contentInfo` data: 
![untitled](https://user-images.githubusercontent.com/21238050/32392201-b16fdd12-c0e5-11e7-9ac9-7efef3a990ed.png)

When I don't have any errors:
<img width="954" alt="screenshot_4" src="https://user-images.githubusercontent.com/21238050/32392259-f31ab4d0-c0e5-11e7-82b1-dba7b9eca418.png">

Probably I still missed something, or I should override `errorPolicy` by another way?

**Comment by edraitsev** [comment]
Any updates about it?

**Comment by noeljackson** [comment]
I can add some more information to this. `data.error` is returned `undefined` when using `errorPolicy: 'all'` Only after removing `errorPolicy` at all was I able to get errors passing through.

**Comment by edraitsev** [comment]
@noeljackson, the problem is not that `data.error` is undefined - I don't get data section when i use `errorPolicy: all`. It works absolutely the same as any other `errorPolicy` value. Seems like it's even not affected my query

![image](https://user-images.githubusercontent.com/21238050/32942614-996ad2c0-cb9a-11e7-909a-2f841eec5392.png)

If I executed the same query with GraphiQL I have error and data sections, but in Apollo I can't see data section

![image](https://user-images.githubusercontent.com/21238050/32942682-cdaf6ab4-cb9a-11e7-8249-0e6028955cb6.png)

**Comment by n0ne** [comment]
@noeljackson I have the same... very strange...

**Comment by VetoS88** [comment]
I have an identical problem. Did anyone get to know how to fetch errors and data from the responce?

**Comment by bradwoods** [comment]
I'm having the same issue

**Comment by stantoncbradley** [comment]
may be related to this bug https://github.com/apollographql/react-apollo/issues/1389

**Comment by hwillson** [comment]
For full error handling details, please see the [Error handling](https://www.apollographql.com/docs/react/features/error-handling/) section of the docs. If anything isn't clear in the docs (or isn't working), please let us know in a new issue. Thanks!
